File: routeMap/app/src/main/java/ru/android/zheka/gmapexample1/AlertDialog.java
package ru.android.zheka.gmapexample1;

import android.app.Dialog;
import android.os.Bundle;
import android.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import ru.android.zheka.gmapexample1.R;
import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.DialogInterface.OnClickListener;

public class AlertDialog extends DialogFragment {
	public String msg;
	private TextView msgField;
	
	public AlertDialog(String msg){
		this.msg = msg;
	}
	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState){		
		System.out.println("start onCreateDialog");
		super.onCreate(savedInstanceState);
		LayoutInflater inflater = getActivity().getLayoutInflater();
		System.out.println("start getting builder");
		android.app.AlertDialog.Builder builder = new android.app.AlertDialog.Builder(getActivity());
		System.out.println("start setView");
		View view = inflater.inflate(R.layout.alert_dialog, null);
		builder.setView(view);		
		System.out.println("start getting nameField,builder is "+builder);
		//builder.setView(R.layout.save_dialog);
		msgField = (TextView)(view.findViewById(R.id.text));
		msgField.setText(msg);
		System.out.println("start setPositiveButton");
		builder.setPositiveButton(R.string.ok_save_point, new OnClickListener() {
			@Override
			public void onClick(DialogInterface arg0, int arg1) {
				arg0.cancel();
			}
		});
		System.out.println("start returning");
		return builder.create();
	}
}

File: routeMap/app/src/main/java/ru/android/zheka/gmapexample1/AlertDialog.kt
package ru.android.zheka.gmapexample1

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.os.Bundle
import android.view.View
import android.widget.TextView

class AlertDialog(var msg: String) : DialogFragment() {
    private var msgField: TextView? = null
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        println("start onCreateDialog")
        super.onCreate(savedInstanceState)
        val inflater = activity.layoutInflater
        println("start getting builder")
        val builder = AlertDialog.Builder(activity)
        println("start setView")
        val view = inflater.inflate(R.layout.alert_dialog, null)
        builder.setView(view)
        println("start getting nameField,builder is $builder")
        //builder.setView(R.layout.save_dialog);
        msgField = view.findViewById<View>(R.id.text) as TextView
        msgField!!.text = msg
        println("start setPositiveButton")
        builder.setPositiveButton(R.string.ok_save_point) { arg0, arg1 -> arg0.cancel() }
        println("start returning")
        return builder.create()
    }

}